### ISSUE TITLE
API.md empty

### ISSUE THREAD

**Issue by corrjo** [cdklabs/jsii-docgen#415]
I have a projen construct library, and at some point I upgraded packages and my API.md turned into just the title
 `# API Reference <a name="API Reference"></a>`
and 5 blank lines. 

Even after updating to latest I am still unable generate an API.md

This is the change that seemed to cause the issue:
[yarnlockdiff.txt](https://github.com/cdklabs/jsii-docgen/files/7063674/yarnlockdiff.txt)

```
   "devDependencies": {
-    "@aws-cdk/assert": "1.116.0",
+    "@aws-cdk/assert": "1.117.0",
     "@types/jest": "^26.0.24",
     "@types/node": "^10.17.0",
-    "@typescript-eslint/eslint-plugin": "^4.28.4",
-    "@typescript-eslint/parser": "^4.28.4",
-    "esbuild": "^0.12.15",
-    "eslint": "^7.31.0",
+    "@typescript-eslint/eslint-plugin": "^4.29.0",
+    "@typescript-eslint/parser": "^4.29.0",
+    "esbuild": "^0.12.18",
+    "eslint": "^7.32.0",
     "eslint-config-prettier": "^8.3.0",
     "eslint-import-resolver-node": "^0.3.4",
     "eslint-import-resolver-typescript": "^2.4.0",
@@ -45,104 +42,103 @@
     "eslint-plugin-prettier": "^3.4.0",
     "jest": "^27.0.6",
     "jest-junit": "^12",
-    "jsii": "^1.31.0",
-    "jsii-diff": "^1.31.0",
-    "jsii-docgen": "^1.8.110",
-    "jsii-pacmak": "^1.31.0",
-    "jsii-release": "^0.2.67",
+    "jsii": "^1.32.0",
+    "jsii-diff": "^1.32.0",
+    "jsii-docgen": "^3.2.19",
+    "jsii-pacmak": "^1.32.0",
+    "jsii-release": "^0.2.82",
     "json-schema": "^0.3.0",
     "npm-check-updates": "^11",
     "prettier": "^2.3.2",
-    "projen": "^0.17.95",
-    "standard-version": "^9",
+    "projen": "^0.27.18",
     "ts-jest": "^27.0.4",
     "typescript": "^4.3.5"
   }```

**Comment by eladb** [comment]
assigning to  @iliapolo

**Comment by corrjo** [comment]
@iliapolo Any ideas or pointers to places I could dig?

**Comment by iliapolo** [comment]
@corrjo can you point me to your source code?

**Comment by Chriscbr** [comment]
@webratz Can you provide a reproducible example or link to a repo perhaps?

I wonder if there's some kind of bug obtaining the JSII metadata.

**Comment by webratz** [comment]
Its sadly a non public project, but i can try tomorrow to build an example.
I think the important part is, that there is no standard root level export but sth like that
```ts
export * as ecr from './constructs/ecr';
```

**Comment by austinwu** [comment]
I ran into the same issue in a few private projects, and also found that `API.md` misses any "submodule exports". I attempted to follow the same directory structure as [projen](https://github.com/projen/projen/blob/main/src/index.ts), so I found it surprising that it didn't work as expected.

Here's a toy example: https://github.com/austinwu/projen-issues/pull/1

**Comment by Chriscbr** [comment]
Ahh, I see this is all because we don't generate docs in the root module by default. 🤦  So this is probably a symptom of https://github.com/cdklabs/jsii-docgen/issues/540

**Comment by Chriscbr** [comment]
Hey @corrjo @webratz @austinwu - I believe this should be fixed in `jsii-docgen@6.0.0`. I'm closing this out, but feel free to @ me or create a new issue if you still run into problems. Thanks 🙂
